pallene/coder.lua

```diff
@@ -390,7 +384,7 @@ function Coder:pallene_entry_point_declaration(f_id)
     local args = {} -- { {ctype, name , comment} }
     table.insert(args, {"lua_State *" , "L",    ""})
     table.insert(args, {"StackValue *", "base", ""})
-    table.insert(args, {"Udata *"     , "G",    ""})
+    table.insert(args, {"Udata *"     , "K",    ""})
```

Comment:
> I wonder if the K could also be a Tvalue * restrict

I think it would have to be a `UValue * restrict` instead? The `Udata` struct doesn't seem to have a `TValue*` member.

```cpp
typedef struct Udata {
  CommonHeader;
  unsigned short nuvalue;  /* number of user values */
  size_t len;  /* number of bytes */
  struct Table *metatable;
  GCObject *gclist;
  UValue uv[1];  /* user values */
} Udata;

``` 

and then we can access the tvalue ptrs by `K[n]->uv` maybe?

Reply:
> The Udata struct doesn't seem to have a TValue* member.

There's a flexible array at the end of the struct. The pointer we want is `&uv[0]`.

> however it won't be able to access it via U because U is NULL

This is a bigger problem. Let's go with `Udata * restrict` then.

Reply:
Wait, you're right. The `LUA_MAXALIGN` thing in the `UValue` means that the `TValue` array really doesn't exist. We certainly want `Udata * restrict` then.